### PR TITLE
Ensured that rows can be changed and comments added in the Climatic > Examine/Edit > Data entry Dialog

### PR DESCRIPTION
**Pull request by N-thony**
Fixes partially #8281
@Fidel365 @rdstern this partially fixes the bug reported in the corresponding issue when the comment is added or a row values has changed. Have a look.

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony that's excellent, I think one of the problems is fixed.  It now doesn't make that funny extra column.  
However, one critical problem still remains.  And I think I have found it.  I am not sure of the solution though.

After I make the change, I made it to Xai-Xai, and just added a comment.
Then I returned to the dialog.
In the returned dialog I now change the Station name.
Now I find that Ok is enabled - though I have not changed any data.  So it should not be.

I pressed To Script to see what it has produced:

```
# Dialog: Daily Data Editing/Entry

new_data <- data.frame(Station_Name="CHIMOIO")
data_book$save_data_entry_data(data_name="Mozambique_3_stations", new_data=new_data, rows_changed="22951", comments_list=list())
rm(new_data)

```
It has most of this code from last time, when I just changed a comment.  BUT it has a new station, i.e. the one I have now selected.  

I closed the dialog and checked the 3 stations - the 3 factors are fine.

Now I ran the script - so like pressing Ok.  Then I checked the stations again and found one row less in Xai-Xai has changed and become Chimoyo!

(When I hit the problem, as I was working before, I had changed a whole month, and it had changed a month from one station into the other.

I checked back in the dialog and when I change the variables to view it also enables Ok.  So that is currently another risky thing to do.  

I don't actually mind Ok being enabled, when there is nothing to change, as long as it then doesn't change anything!  But I suggest it would be simpler to only enable Ok, when there is at least a comment to change, or data to change, or both.

Here are the data I have been using.

[Mozambique_3_stations_shifted_summaries.zip](https://github.com/africanmathsinitiative/R-Instat/files/12458358/Mozambique_3_stations_shifted_summaries.zip)

**Review by rdstern — APPROVED** [review]
Yippee!  At last this now seems fixed.  Many thanks @N-thony.  @lloyddewit please check and merge as soon as you can.  This is useful for the forthcoming Zambia workshop.

File: instat/static/InstatObject/R/data_object_R6.R [hunk 1]
```diff
@@ -4315,6 +4315,7 @@ DataSheet$set("public", "get_data_entry_data", function(station, date, elements,
 })
 
 DataSheet$set("public", "save_data_entry_data", function(new_data, rows_changed, add_flags = FALSE, ...) {
+  if (ncol(new_data) > 1) {
```
Comment (lloyddewit) [inline instat/static/InstatObject/R/data_object_R6.R:4318]: indent lines 4319 to 4345?

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@N-thony thanks look good, just one small formatting suggestion

**Review by rdstern — DISMISSED** [review]
Still looks fine to me

File: instat/static/InstatObject/R/data_object_R6.R [hunk 2]
```diff
@@ -4315,33 +4315,35 @@ DataSheet$set("public", "get_data_entry_data", function(station, date, elements,
...
+        }
+      }
+    }
+    if(length(nrow(new_data)) > 0) cat("Row(s) updated: ", nrow(new_data), "\n")
+    self$set_data(changed_data)
+    # Added this line to fix the bug of having the variable names in the metadata changing to NA
+    # This affects factor columns only  - we need to find out why and how to solve it best
+    self$add_defaults_variables_metadata(self$get_column_names())
+    self$data_changed <- TRUE
+   }
```
Comment (lloyddewit) [inline instat/static/InstatObject/R/data_object_R6.R:4346]:
```suggestion
  }
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**